### DISCUSSION TITLE
[Proposal] Arrow operator as shorthand return for single line if statements

### DISCUSSION THREAD

**Discussion by murshex** [dotnet/csharplang#6384] [Language Ideas]
Example:

```csharp
public static string FindCollectionName(Type type)
{
    if (typeof(Message).IsAssignableFrom(type))
        return "Messages";
    
    if (typeof(UserData).IsAssignableFrom(type))
        return "UserData";

    return DocumentConventions.DefaultGetCollectionName(type);
}
```

Could be writen as:

```csharp
public static string FindCollectionName(Type type)
{
    if (typeof(Message).IsAssignableFrom(type)) => "Messages";
    if (typeof(UserData).IsAssignableFrom(type)) => "UserData";

    return DocumentConventions.DefaultGetCollectionName(type);
}
```

What do you think? Suddenly thought about this and I guess the syntax would be nice.

**Comment by HaloFour** [comment]
```cs
if (typeof(Message).IsAssignableFrom(type)) return "Messages";
```

I don't understand what the `=>` has to do with an implied `return` from the method.  That's not how the operator works in any other context in the language.

**Reply by murshex** [reply]
Thats not correct. Here are a few examples:

```csharp
public int TestMethod() => 1;
public int TestProperty => 2; // Get-only property
```

I think there should be other examples too

**Reply by HaloFour** [reply]
Right, in those cases the operator specifies that the member itself is a single expression, so it has to return a value.  An `if` statement is not a member.

**Comment by erik-kallen** [comment]
I can't imagine it happening, but it would be kind of nice to be able to use if as an expression (and avoid the ?: operator), something like

```
public static string FindCollectionName(Type type)
{
    return if (typeof(Message).IsAssignableFrom(type)) => "Messages"
           if (typeof(UserData).IsAssignableFrom(type)) => "UserData"
           else DocumentConventions.DefaultGetCollectionName(type);
}
```

**Reply by HaloFour** [reply]
I don't think it's bad using the ternary operator either.  IMO it's quite readable if you format it over multiple lines:

```cs
public static string FindCollectionName(Type type) =>
        typeof(Message).IsAssignableFrom(type) ? "Messages"
        : typeof(UserData).IsAssignableFrom(type) ? "UserData"
        : DocumentConventions.DefaultGetCollectionName(type);
```

**Reply by erik-kallen** [reply]
It's not too bad (which is why I don't think it needs to be changed), but it IMO doesn't read nearly as nice as the if-as-expression version that is used in Rust or Scala. And I say this as someone who has done C-style languages in all of my career.

**Reply by bernd5** [reply]
You could use a swtich expression, too:

```csharp
static string FindCollectionName(Type type) =>
    type switch{
     _ when typeof(Message).IsAssignableFrom(type) => "Messages",
     _ when typeof(UserData).IsAssignableFrom(type) => "UserData",
     _ => DocumentConventions.DefaultGetCollectionName(type)       
    };
```

**Comment by Rekkonnect** [comment]
I would be more interested in `if` expressions rather than this; see how some other languages handle `if` expressions (Kotlin, Python are some examples if I'm not mistaken).
